### PR TITLE
Move Streamly.FileSystem.Handle to streamly-core

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes: https://github.com/composewell/streamly/issues/1834

**Comment by rnjtranjan** [comment]
fixes:https://github.com/composewell/streamly/issues/1836
https://github.com/composewell/streamly/issues/1835

File: core/streamly-core.cabal [hunk 1]
```diff
@@ -358,12 +381,15 @@ library
...
                      , deepseq           >= 1.4.1 && < 1.5
+                     --, directory         >= 1.2.2 && < 1.4
                      , exceptions        >= 0.8   && < 0.11
                      , ghc-prim          >= 0.2   && < 0.10
                      , mtl               >= 2.2   && < 2.3
                      , transformers      >= 0.4   && < 0.7
+                     , template-haskell  >= 2.13 && < 2.20
 
                      , fusion-plugin-types >= 0.1 && < 0.2
+                     , unicode-data      >= 0.1   && < 0.4
```
Comment (harendra-kumar) [inline core/streamly-core.cabal:392]: We cannot depend on unicode-data. Only boot libraries.

**Comment by adithyaov** [comment]
Need to fix packdiff and bench-report before we can merge this

**Comment by harendra-kumar** [comment]
These three should be done in separate PRs for better reviewability, and in the following order:

* Unicode modules (except the ones that depend on unicode-data e.g. Utf8)
* FileSystem.Handle module
* Console.Stdio module

**Comment by rnjtranjan** [comment]
New PRs has been created:
https://github.com/composewell/streamly/pull/1853
https://github.com/composewell/streamly/pull/1854
https://github.com/composewell/streamly/pull/1855

**Pull request closed without merging**